coralproject/cay
# Implement design standards

We were working closely with @egrdina and @IAmSamHankins on the styleguide and the UI Components for Coral and we believe it's a good idea to have our own components to ease the design/develop experience
## What's new with this PR?

Implementing Design standards for the following issues
https://github.com/coralproject/ask/issues/129
https://github.com/coralproject/ask/issues/128
https://github.com/coralproject/ask/issues/127
### New UI Components
### CoralButton

Params
- type (@String)
  
  >    `success`, `default`, `primary`. and also by color (`violet`, `black`, `etc`) . We are working with @egrdina and @IAmSamHankins  to bring more types!
- active (@Boolean)
  
  >  Pass `true`, `false` or a condition
- customColor
  
  > (@String) A hex color. This can only be used with `type="custom"`. This feature builds a button style given a certain color.
- icon
  
  > (@String). The name of the Material Icon 
- disabled
  
  > (@Boolean)
- onClick, style, className, etc.
  
  > Normal usage. Passing props to the child component is transparent.
### Usage:

``` javascript

import { CoralButton } from '../components/ui';

{/* Basic Usage*/}

<CoralButton
   type="success"
   onClick={saveItem}
   icon="done"
>
  Save
</CoralButton>
```

``` javascript
{/* "Complex" Usage*/}

<CoralButton
   className="form-preview-button"
   type="primary"
   icon="visibility"
   onClick={onOpenPreview}
   active={flagged}
   style={{ width: 150, marginRight: 10 }}
>
  Preview
</CoralButton>
```

``` javascript
{/* Custom Color Usage*/}

<CoralButton
   type="custom"
   customColor={settings.color}
   onClick={onOpenPreview}
>
  Flagged
</CoralButton>

```
### CoralIcon

Renders an Icon using the Material Icon Font

Params
- icon (@String)
  > Pass the name of the Icon. Find them here: [https://design.google.com/icons/](https://design.google.com/icons/)

``` javascript
import { CoralIcon } from '../components/ui';

<CoralIcon icon="done" />

```
### CoralIconButton

Renders an Icon using the Material Icon Font

Params
- icon (@String)
  
  > Pass the name of the Icon. Find them here: [https://design.google.com/icons/](https://design.google.com/icons/)
- disabled
  
  > (@Boolean)

``` javascript
import { CoralIconButton } from '../components/ui';

<CoralIconButton icon="content_copy" onClick={ () => this.handleClick(position) } />

```
## You can import multiple components

``` javascript
import { CoralIconButton,  CoralButton} from '../components/ui';
```
## Thoughts

Radium is not friendly with stateless components. 
## Changelog
- Added Roboto Font Offline 
- Removed old fonts, and usages of FontAwesome, and GlyphI
- Material Font Icons Added Offline
- 3 Coral Components Created

**Share your feedback**
Please feel free to share your feedback and thoughts about this. 😄 
## How do I test this PR?
- Checkout this branch
- Check if the buttons are working as expected (Editor, Submissions)
- See if they match the Styleguide
  [Coral.MDL.Styles.pdf](https://github.com/coralproject/cay/files/499151/Coral.MDL.Styles.pdf)

This is a work in progress
## TODO:
- [x] Use settings theme
- [x] Proptypes
- [ ] Add Coral Card Component
- [x] Add full test coverage
- [x] Expand API for Button Effects
- [x] Add support for sizes and icon sizes
- [ ] Do an online demo with component-playground

@coralproject/frontend
